### PR TITLE
Switch to vcpkg for NetHook2 dependencies

### LINKED ISSUE #1108: [NetHook2]: Fails to build with VS2022 toolchain
### What did you expect to happen?

I expected that NetHook2 compiles with VS2022

### Instead of that, what actually happened?

```
"C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2.sln" (default target) (1) ->
"C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj" (default target) (2) ->
(Link target) ->
  libprotobufd.lib(any.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_and_abort
@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(extension_set_heavy.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate
_names_and_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(implicit_weak_message.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alterna
te_names_and_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(text_format.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_a
nd_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(dynamic_message.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_nam
es_and_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(reflection_ops.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_name
s_and_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(wire_format.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_a
nd_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(descriptor.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_an
d_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(map_field.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_and
_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(message_lite.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_
and_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(message.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_and_a
bort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(arenastring.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_alternate_names_a
nd_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  libprotobufd.lib(generated_message_reflection.obj) : error LNK2001: unresolved external symbol ___std_init_once_link_
alternate_names_and_abort@0 [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]
  C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\Debug\NetHook2.dll : fatal error LNK1120: 1 unresolved exter
nals [C:\Users\Yaakov\source\repos\SteamKit\Resources\NetHook2\NetHook2\NetHook2.vcxproj]

    15 Warning(s)
    14 Error(s)
```

### Relevant log output

_No response_

### Example Code

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by psychonic**
This change switches from NetHook2's dependencies from manual download scripts to using vcpkg. With that, there are other minor changes to versions of things to make the transition a bit more smooth, avoid issues of package unavailability or incompatibility. The build toolset has also been upgraded from Visual Studio 2019 to Visual Studio 2022.

The build has been tested and works the same as usual.

Dependency updates included are below. Versions are pinned in the new `vcpkg.json`.

* Detours: `2.1.0` -> `4.0.1`
* Protobuf: `3.15.6` -> `3.15.8`
* Zlib: `1.2.12` -> `1.2.13`

Fixes #1108

### CODE REVIEW

File: Resources/NetHook2/readme.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ These messages are dumped to file, and can be analyzed further with NetHookAnaly
 #### Building
 
 1. Execute `SetupDependencies.cmd` to automatically acquire and build the zlib and protobuf headers and libraries. You will need CMake for this.
```
Comment (yaakov-h) [inline Resources/NetHook2/readme.md:11]: Is CMake still necessary?

Comment (psychonic) [reply Resources/NetHook2/readme.md]: Nope! (at least not directly). I've updated that section now. Good catch

**Review by xPaw — APPROVED** [review]
Tested dll from CI, works for me.

**Comment by yaakov-h** [comment]
Does `detours64.lib` get to stay around or should that be deleted now?

File: Resources/NetHook2/SetupDependencies.cmd [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+    )
+
+    if exist "%InstallDir%\Common7\Tools\vsdevcmd.bat" (
+        call "%InstallDir%\Common7\Tools\vsdevcmd.bat"
+    )
+)
+
+where.exe /q vcpkg
+if %ERRORLEVEL%==1 (
+    if not exist "vcpkg/" (
```
Comment (yaakov-h) [inline Resources/NetHook2/SetupDependencies.cmd:15]: I don't like this, can't we ask the user to have vcpkg installed?
Or is this for CI?aaa

**Comment by psychonic** [comment]
> Does `detours64.lib` get to stay around or should that be deleted now?

Gone.

File: Resources/NetHook2/SetupDependencies.cmd (hunk 2 above)
Comment (psychonic) [reply Resources/NetHook2/SetupDependencies.cmd:15]: This is mostly for CI. For what it's worth, this copy of it only gets added into PATH for the duration of the session.

Comment (psychonic) [reply Resources/NetHook2/SetupDependencies.cmd]: I now realize that it does a full setup into the user's profile despite the binary living under the project folder. (This is my first time with vcpkg). That's indeed not ideal. I've updated it now to explicitly only do that for CI, else print out instructions for where to get it if missing.

File: Resources/NetHook2/SetupDependencies.cmd [hunk 3]
```diff
@@ -12,18 +12,13 @@ if not defined DevEnvDir (
 
 where.exe /q vcpkg
 if %ERRORLEVEL%==1 (
-    if not exist "vcpkg/" (
+    if defined GITHUB_ACTIONS (
```
Comment (xPaw) [inline Resources/NetHook2/SetupDependencies.cmd:15]: Simply `CI` would be more future proof.

Comment (psychonic) [reply Resources/NetHook2/SetupDependencies.cmd]: Good call

**Pull request merged**